pkg/plugins/pg/source/source.go

```diff
@@ -68,17 +64,19 @@ type Source struct {
 	sub *pgoutput.Subscription
 	// subErr is the return value of the `sub.Start` function in withCDC
 	subErr error
-	// buffer holds a reference to our CDCIterator so that we can read
-	// from the buffer when rows are no longer available.
-	buffer *CDCIterator
+	// cdc holds a reference to our CDCIterator so that we can read
+	// from the cdc when rows are no longer available.
+	cdc Iterator
```

Comment:
If this should always be a CDCIterator, should we declare it as such?

Reply:
I don't like the stutter that introduces, unless you mean making it a *CDCIterator instead of an Iterator? If it's the latter, I would rather keep it as the interface.

Reply:
Yup, I meant making it a *CDCIterator instead of an Iterator, since it will always be that, and actually has to be that. But I don't feel too strongly on that.